### PR TITLE
Prevent safecracking attempt while wearing earbuds

### LINKED ISSUE #36013: Can crack a safe with a stethoscope while simultaneously listening to an mp3 player
Game version: 0.D-10020-gd9d23c5 (tiles), build 9981
Operating system: Windows 10

Mods active: None

### Expected behavior
A prompt to turn off your mp3 player before trying to crack a safe with a stethoscope, or just a message saying something like: "There's too much noise for you to try to crack the safe!"

### Actual behavior
The entire time spent cracking the safe, music just plays and focus goes up like you aren't doing anything but listening to music.

### Steps to reproduce the behavior
Find stethoscope and mp3 player, turn on mp3 player, crack safe.

### PR DESCRIPTION
**Pull request by wapcaplet**
#### Summary
```SUMMARY: Bugfixes "Prevent safecracking attempt while wearing earbuds"```

#### Purpose of change
Fixes #36013 

#### Describe the solution
My solution is two-fold:
- Change the mp3 player (de-)activation to take 1 second of time, instead of 0 seconds
- Check if player has `effect_earphones` in the safe-cracking function `iexamine::safe`; if so deny the attempt with a message: "You can't crack a safe with earbuds on!"

#### Describe alternatives you've considered
I considered leaving mp3 activation time alone, but that would leave a loophole allowing the safecracking attempt to begin immediately after starting the mp3 player. Causing it to take 1 second solved the loophole while also being more realistic.

While I was at it, I considered adding logic to check for for `p.worn_with_flag( "PARTIAL_DEAF" )` but those appear to be fairly rare specialty items that filter loud sounds rather than muting them entirely. Anyway it's outside the scope of this bugfix.

#### Testing
Tested in a debug world - got lucky and found a private resort with a couple of safes to test on:

- Try to crack safe without any tools - "You mess with the dial for a little bit"
- Spawn mp3 player and stethoscope
- Examine safe: "Crack safe?" (OK)
- Activate mp3 player
- Examine safe: "You can't crack a safe with earbuds on!" (OK)
- Deactivate mp3 player
- Wait 1 second for mp3 player to turn off
- Examine safe: "Crack safe?" (OK)

#### Additional context

The mp3 player's 1-turn delay on activation, and 2-turn delay on deactivation has been clearly visible in-game in the experimental branch for quite some time, using the DeadPeople tileset. You may have noticed after turning on an mp3 player that it takes 1 further second for the icon to appear above your avatar, and that after turning it off, 2 seconds must elapse before the icon disappears.

Now, because of the 1-second (de-)activation time, turning it on gives the icon right away, but turning it off still takes 1 more second before the icon goes away (and you can attempt the safe crack again - hence the extra "Wait 1 second" step above).

### CODE REVIEW

File: src/iuse.cpp [hunk 1]
```diff
@@ -4106,8 +4106,10 @@ int iuse::mp3( player *p, item *it, bool, const tripoint & )
         p->add_msg_if_player( m_info, _( "You put in the earbuds and start listening to music." ) );
         if( it->typeId() == "mp3" ) {
             it->convert( "mp3_on" ).active = true;
+            p->moves -= to_moves<int>( 1_seconds );
```
Comment (codemime) [inline src/iuse.cpp]: Please use `Creature::mod_moves` instead.

Comment (wapcaplet) [reply src/iuse.cpp]: OK, would this be equivalent? `p->mod_moves( -100 )`

Comment (codemime) [reply src/iuse.cpp]: Yes, but you can still write it like this: `p->mod_moves( -to_moves<int>( 1_seconds ) )`. I guess we should make an overloaded version of `mod_moves` which accepts `time_duration` directly, but doesn't allow granting moves. Maybe even rename it to something like `consume_moves( 1_second )`.

**Comment by wapcaplet** [comment]
Alright, I pushed one last change, increasing the mp3 on/off time to 2 seconds. This makes the status indicator above your player update correctly after deactivating the mp3 player, and allows a safecrack attempt immediately afterward (as it should).

2 seconds to turn an mp3 player on or off seems pretty generous, even if you assume you're keeping the ear buds in all the time. Heck, I'll spend 5 minutes just trying to figure out what to listen to.

File: src/iuse.cpp [hunk 2]
```diff
@@ -4106,8 +4106,10 @@ int iuse::mp3( player *p, item *it, bool, const tripoint & )
         p->add_msg_if_player( m_info, _( "You put in the earbuds and start listening to music." ) );
         if( it->typeId() == "mp3" ) {
             it->convert( "mp3_on" ).active = true;
+            p->mod_moves( -200 );
         } else if( it->typeId() == "smart_phone" ) {
             it->convert( "smartphone_music" ).active = true;
+            p->mod_moves( -200 );
         }
```
Comment (Night-Pryanik) [inline src/iuse.cpp]: Nitpick, but you could replace two lines of `mod_moves`, one for mp3 and one for smartphone, with one, somewhere after line 4105.

Comment (wapcaplet) [reply src/iuse.cpp]: Due to the lack of any “else” condition in either of these, I though it safest to wait only for known devices.

**Pull request merged**